compose/cli/main.py

```diff
@@ -84,6 +85,20 @@ def main():
         sys.exit(1)
 
 
+def log_api_error(e):
+    if 'client is newer than server' in e.explanation:
+        # we need JSON formatted errors. In the meantime...
+        client_version = e.explanation.split('client API version: ')[1].split(',')[0]
```

Comment:
Indeed, if we have access to the client object that sent the request, then we can just pull the `api_version` from here. 
Just calling `docker_client()` won't help because we need the config version as a parameter. 
I don't know about adding client to `APIError`, isn't that a bit strange considering those errors are always raised by the client itself? It's a reference the user should already have access to. Also if we choose to do so, we'll need to upgrade docker-py as a result.